### PR TITLE
Switch to subtree

### PR DESCRIPTION
**Pull request by ced0ps**
Updated to latest release 2.0.0-rc.2
Fixed subtree upgrade tracking of charts directory

### CODE REVIEW

File: helm/kong-app/UPGRADE.md [hunk 1]
```diff
@@ -56,6 +57,43 @@ text ending with `field is immutable`. This is typically due to a bug with the
 If you encounter this error, deleting any existing `init-migrations` jobs will
 clear it.
 
+## 2.0.0-rc1
+
+### Support for Helm 2 dropped
```
Comment (ubergesundheit) [inline helm/kong-app/UPGRADE.md:62]: `apiVersion` in the `Chart.yaml` file still says `v1`. I'm unsure if we should bump the `apiVersion` as well.

**Comment by ubergesundheit** [comment]
Can you point out the changes from upstream and the ones done by us? 

Also seems like the `values.schema.json` needs an update. The template-app contains a really useful [workflow](https://github.com/giantswarm/template-app/blob/master/.github/workflows/check_values_schema.yaml) to check this in CI.

**Comment by ced0ps** [comment]
> Can you point out the changes from upstream and the ones done by us?
> 
> Also seems like the `values.schema.json` needs an update. The template-app contains a really useful [workflow](https://github.com/giantswarm/template-app/blob/master/.github/workflows/check_values_schema.yaml) to check this in CI.

just did that :)

File: helm/kong-app/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,59 @@
...
+
+* Generated admission webhook certificates now include SANs for compatibility
+  with Go 1.15 controller builds.
+  ([#312](https://github.com/Kong/charts/pull/312)).
+
+### Documentation
+
+* Clarified use of `terminationGracePeriodSeconds`.
+  ([#302](https://github.com/Kong/charts/pull/302))
+
```
Comment (ced0ps) [inline helm/kong-app/CHANGELOG.md:56]: upstream change

File: helm/kong-app/README.md [hunk 3]
```diff
@@ -14,17 +14,12 @@ This chart bootstraps all the components needed to run Kong on a
 $ helm repo add kong https://charts.konghq.com
 $ helm repo update
 
-# Helm 2
-$ helm install kong/kong
-
-# Helm 3
-$ helm install kong/kong --generate-name --set ingressController.installCRDs=false
+$ helm install kong/kong --generate-name
```
Comment (ced0ps) [inline helm/kong-app/README.md:17]: upstream change

File: helm/kong-app/README.md [hunk 4]
```diff
@@ -14,17 +14,12 @@ This chart bootstraps all the components needed to run Kong on a
...
 ```
 
 ## Table of contents
 
 - [Prerequisites](#prerequisites)
-- [Helm 2 vs Helm 3](#important-helm-2-vs-helm-3)
```
Comment (ced0ps) [inline helm/kong-app/README.md:27]: upstream change

File: helm/kong-app/README.md [hunk 5]
```diff
@@ -36,7 +31,7 @@ $ helm install kong/kong --generate-name --set ingressController.installCRDs=fal
   - [Separate admin and proxy nodes](#separate-admin-and-proxy-nodes)
   - [Standalone controller nodes](#standalone-controller-nodes)
   - [Hybrid mode](#hybrid-mode)
-  - [CRDs only](#crds-only)
+  - [CRD management](#crd-management)
```
Comment (ced0ps) [inline helm/kong-app/README.md:34]: upstream change

File: helm/kong-app/README.md [hunk 6]
```diff
@@ -64,30 +59,6 @@ $ helm install kong/kong --generate-name --set ingressController.installCRDs=fal
...
-You can do so either by passing in a custom `values.yaml`
-(`-f` when running helm)
-or by passing `--set ingressController.installCRDs=false`
-at the command line.
-
-**If you do not set this value to `false`, the helm chart will not install correctly.**
-
-Use helm CLI flag `--skip-crds` with `helm install` if you want to skip
-CRD creation while creating a release.
-
```
Comment (ced0ps) [inline helm/kong-app/README.md:90]: upstream change

File: helm/kong-app/README.md [hunk 7]
```diff
@@ -96,10 +67,6 @@ To install Kong:
 $ helm repo add kong https://charts.konghq.com
 $ helm repo update
 
-# Helm 2
-$ helm install kong/kong
-
-# Helm 3
```
Comment (ced0ps) [inline helm/kong-app/README.md:102]: ok basically all deletes are upstream changes

File: helm/kong-app/README.md [hunk 8]
```diff
@@ -427,17 +394,37 @@ documentation on Service
...
-
-For Helm 2 installations, CRDs are managed as part of a release, and are
-deleted if the release is. This can cause issues for clusters with multiple
-Kong installations, as one release must remain in place for the rest to
-function. To avoid this, you can create a CRD-only release by setting
-`deployment.kong.enabled: false` and `ingressController.enabled: false`.
-
-On Helm 3, CRDs are created if necessary, but are not managed along with the
-release. Releases can be deleted without affecting CRDs; CRDs are only removed
-if you delete them manually.
```
Comment (ced0ps) [inline helm/kong-app/README.md:440]: yup, this one too

File: helm/kong-app/README.md [hunk 9]
```diff
@@ -427,17 +394,37 @@ documentation on Service
...
+
+While Helm 3's CRD management system is recommended, there is no simple means
+of migrating away from release-managed CRDs if you previously installed your
+release with the old system (you would need to back up your existing custom
+resources, delete your release, reinstall, and restore your custom resources
+after). As such, the chart detects if you currently use release-managed CRDs
+and continues to use the old CRD templates when using chart version 2.0+. If
+you do (your resources will have a `meta.helm.sh/release-name` annotation), we
+_strongly_ recommend that you back up all associated custom resources in the
+event you need to recover from unintended CRD deletion.
```
Comment (ced0ps) [inline helm/kong-app/README.md:427]: upstream change

File: helm/kong-app/README.md [hunk 10]
```diff
@@ -461,7 +448,7 @@ directory.
 | Parameter                          | Description                                                                           | Default             |
 | ---------------------------------- | ------------------------------------------------------------------------------------- | ------------------- |
 | image.repository                   | Kong image                                                                            | `kong`              |
-| image.tag                          | Kong image version                                                                    | `2.2`               |
+| image.tag                          | Kong image version                                                                    | `2.0`               |
```
Comment (ced0ps) [inline helm/kong-app/README.md:451]: upstream change (back?)

File: helm/kong-app/README.md [hunk 11]
```diff
@@ -564,7 +551,7 @@ section of `values.yaml` file:
 | image.tag                          | Version of the ingress controller                                                     | 0.9.1                                                                        |
 | readinessProbe                     | Kong ingress controllers readiness probe                                              |                                                                              |
 | livenessProbe                      | Kong ingress controllers liveness probe                                               |                                                                              |
-| installCRDs                        | Create CRDs. **FOR HELM3, MAKE SURE THIS VALUE IS SET TO `false`.**  Regardless of value of this, Helm v3+ will install the CRDs if those are not present already. Use `--skip-crds` with `helm install` if you want to skip CRD creation.                 | true                                                                         |
+| installCRDs                        | Creates managed CRDs.                                                                 | false
```
Comment (ced0ps) [inline helm/kong-app/README.md:554]: upstream change

File: helm/kong-app/README.md [hunk 12]
```diff
@@ -594,8 +581,9 @@ For a complete list of all configuration values you can set in the
 | readinessProbe                     | Kong readiness probe                                                                  |                     |
 | livenessProbe                      | Kong liveness probe                                                                   |                     |
 | lifecycle                          | Proxy container lifecycle hooks                                                       | see `values.yaml`   |
-| terminationGracePeriodSeconds      | Related to lifecycle hook                                                             | 30                  |
+| terminationGracePeriodSeconds      | Sets the [termination grace period](https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/#hook-handler-execution) for Deployment pods | 30                  |
 | affinity                           | Node/pod affinities                                                                   |                     |
+| topologySpreadConstraints          | Control how Pods are spread across cluster among failure-domains                      |                     |
```
Comment (ced0ps) [inline helm/kong-app/README.md:586]: upstream change

File: helm/kong-app/README.md [hunk 13]
```diff
@@ -676,9 +664,12 @@ configuration can be placed under the `.env` key.
 
 #### Kong Enterprise License
 
-All Kong Enterprise deployments require a license. If you do not have a copy
-of yours, please contact Kong Support. Once you have it, you will need to
-store it in a Secret:
+Kong Enterprise 2.3+ can run with or without a license. If you wish to run 2.3+
+without a license, you can skip this step and leave `enterprise.license_secret`
```
Comment (ced0ps) [inline helm/kong-app/README.md:668]: upstream change

File: helm/kong-app/README.md [hunk 14]
```diff
@@ -676,9 +664,12 @@ configuration can be placed under the `.env` key.
...
+unset. Earlier versions require a license.
+
+If you have paid for a license, but you do not have a copy of yours, please
+contact Kong Support. Once you have it, you will need to store it in a Secret:
```
Comment (ced0ps) [inline helm/kong-app/README.md:672]: upstream change

**Comment by ced0ps** [comment]
> Can you point out the changes from upstream and the ones done by us?
> 
> Also seems like the `values.schema.json` needs an update. The template-app contains a really useful [workflow](https://github.com/giantswarm/template-app/blob/master/.github/workflows/check_values_schema.yaml) to check this in CI.

@ubergesundheit I started marking changes but then realised that all changes are upstream since the ones we did are persistent with current master.

**Comment by ubergesundheit** [comment]
Thanks! So this PR just sets up the subtree? I'm a little confused about https://github.com/giantswarm/kong-app/pull/111#discussion_r607913393. IMO we either should follow upstream and make this chart use `apiVersion: v2` or add an explicit entry to our changelog that we still support helm 2 (if possible).  I understand the changelog in this PR is an upstream one.

File: helm/kong-app/UPGRADE.md (hunk 1 above)
Comment (ced0ps) [reply helm/kong-app/UPGRADE.md:62]: @ubergesundheit I see what you mean. That was my oversight. I'll fix it right away.

**Comment by ubergesundheit** [comment]
TODO

- pull the upstream remote

**Review by ubergesundheit — APPROVED** [review]
LGTM

**Pull request merged**